File: 2756-buy-two-chocolates/buy-two-chocolates.cpp
class Solution {
public:
    int buyChoco(vector<int>& prices, int money) {
        priority_queue<int>pq;

        for(int i=0;i<prices.size();i++){
            if(pq.size()<2){
                pq.push(prices[i]);
            }
            else{
                if(pq.top()>prices[i]){
                    pq.pop();
                    pq.push(prices[i]);
                }
            }
        }
        int sum=0;
        sum+=pq.top();
        pq.pop();
       sum+= pq.top();
        return sum<=money?money-sum:money;
    }
};

File: 2756-buy-two-chocolates/buy-two-chocolates.java
class Solution {
    public int buyChoco(int[] prices, int money) {
         PriorityQueue<Integer>pq=new PriorityQueue<>((a,b)->b-a);

        for(int i=0;i<prices.length;i++){
            if(pq.size()<2){
                pq.offer(prices[i]);
            }
            else{
                if(pq.peek()>prices[i]){
                    pq.poll();
                    pq.offer(prices[i]);
                }
            }
        }
        int sum=0;
        sum+=pq.peek();
        pq.poll();
       sum+= pq.peek();
        return sum<=money?money-sum:money;
    }
}